internal/store/group.go

```diff
@@ -182,6 +169,89 @@ func (sqlStore *SQLStore) GetGroupRollingMetadata(groupID string) (*GroupRolling
 	return metadata, nil
 }
 
+// GetGroupStatus returns total number of installations in the group as well as number or
+// Installations already rolled out and awaiting rollout.
+//
+// Note: This function uses the same conditions as GetGroupRollingMetadata to be more accurate
+// with the internal state seen by the Group Supervisor.
+func (sqlStore *SQLStore) GetGroupStatus(groupID string) (*model.GroupStatus, error) {
+	group, err := sqlStore.GetGroup(groupID)
+	if err != nil {
+		return nil, err
+	}
+	if group == nil {
+		return nil, nil
```

Comment:
I'm not sure I understand why the case where the group cannot be found isn't an error. It seems to me like this should return `nil, errors.New(`.

I don't feel strongly about this (1/5), however I'm interested in your thinking regarding returning `nil, nil` here.

Reply:
Generally we don't treat "not found" as an error in the stores since it's a valid success response. For example, I could be running the command to check if a resource exists or not and I wouldn't want it to error on it not being found.

For this case I'm 0/5 since it's trying to pull a group's status and not get a resource directly

Reply:
Yes, that is what I saw in the code.
Although I am not a fan of the pattern with returning `nil, nil`,  I think we should return `404` if querying for the status of a group that does not exist, but I am also 0/5. 
If no one has objections I will leave it as it is.

Reply:
I'm good with as is 👍

Reply:
Sounds good to me

Reply:
+1 from me as well